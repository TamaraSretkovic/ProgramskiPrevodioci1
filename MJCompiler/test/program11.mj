//	Semantika test 
//  - Expr = -Term; Term mora biti int.

program test302
    int d;
    const char c = 'a';
    const bool cond = true;
    int niz1[];
    int niz2[];
{
    void func(int a, char c) {

    }

	void main() {
		d = -c;
	}
}
