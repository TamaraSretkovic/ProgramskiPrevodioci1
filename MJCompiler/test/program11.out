INFO  01:15:56,233 - Compiling source file: /home/tamarica/Documents/tamara/velika skola/8 semestar/PP1/projekat/projecat_2019_2020/MJCompiler/test/program11.mj
INFO  01:15:56,257 - #2 program
INFO  01:15:56,257 - #46 test302
INFO  01:15:56,257 - #46 int
INFO  01:15:56,258 - #46 d
INFO  01:15:56,258 - #5 ;
INFO  01:15:56,259 - #7 const
INFO  01:15:56,259 - #46 char
INFO  01:15:56,259 - #46 c
INFO  01:15:56,259 - #23 =
INFO  01:15:56,259 - #48 a
INFO  01:15:56,260 - #5 ;
INFO  01:15:56,260 - #7 const
INFO  01:15:56,260 - #46 bool
INFO  01:15:56,261 - #46 cond
INFO  01:15:56,261 - #23 =
INFO  01:15:56,261 - #49 1
INFO  01:15:56,261 - #5 ;
INFO  01:15:56,261 - #46 int
INFO  01:15:56,261 - #46 niz1
INFO  01:15:56,261 - #8 [
INFO  01:15:56,261 - #9 ]
INFO  01:15:56,261 - #5 ;
INFO  01:15:56,261 - #46 int
INFO  01:15:56,261 - #46 niz2
INFO  01:15:56,261 - #8 [
INFO  01:15:56,262 - #9 ]
INFO  01:15:56,262 - #5 ;
INFO  01:15:56,262 - #3 {
INFO  01:15:56,262 - #10 void
INFO  01:15:56,262 - #46 func
INFO  01:15:56,262 - #12 (
INFO  01:15:56,262 - #46 int
INFO  01:15:56,263 - #46 a
INFO  01:15:56,263 - #6 ,
INFO  01:15:56,263 - #46 char
INFO  01:15:56,263 - #46 c
INFO  01:15:56,263 - #11 )
INFO  01:15:56,263 - #3 {
INFO  01:15:56,263 - #4 }
INFO  01:15:56,264 - #10 void
INFO  01:15:56,264 - #46 main
INFO  01:15:56,264 - #12 (
INFO  01:15:56,264 - #11 )
INFO  01:15:56,265 - #3 {
INFO  01:15:56,265 - #46 d
INFO  01:15:56,265 - #23 =
INFO  01:15:56,265 - #31 -
INFO  01:15:56,265 - #46 c
INFO  01:15:56,265 - #5 ;
INFO  01:15:56,266 - #4 }
INFO  01:15:56,266 - #4 }
INFO  01:15:56,270 - ==================SINTAKSNO STABLO==================
INFO  01:15:56,271 - Program(
  ProgName(
   test302
  ) [ProgName]
  DeclarationListDeclaration(
    DeclarationListDeclaration(
      DeclarationListDeclaration(
        DeclarationListDeclaration(
          DeclarationListDeclaration(
            NoDeclarationList(
            ) [NoDeclarationList]
            DeclarationListVar(
              VarDeclarationV(
                VarType(
                  Type(
                   int
                  ) [Type]
                ) [VarType]
                VarDecl(
                  Ident(
                   d
                  ) [Ident]
                ) [VarDecl]
              ) [VarDeclarationV]
            ) [DeclarationListVar]
          ) [DeclarationListDeclaration]
          DeclarationListConst(
            ConstDeclarationTypeListSemi(
              VarType(
                Type(
                 char
                ) [Type]
              ) [VarType]
              ConstDecl(
                ConstDeclarationActual(
                 c
                  AssignOp(
                  ) [AssignOp]
                  CharConst(
                   a
                  ) [CharConst]
                ) [ConstDeclarationActual]
              ) [ConstDecl]
            ) [ConstDeclarationTypeListSemi]
          ) [DeclarationListConst]
        ) [DeclarationListDeclaration]
        DeclarationListConst(
          ConstDeclarationTypeListSemi(
            VarType(
              Type(
               bool
              ) [Type]
            ) [VarType]
            ConstDecl(
              ConstDeclarationActual(
               cond
                AssignOp(
                ) [AssignOp]
                BoolConst(
                 1
                ) [BoolConst]
              ) [ConstDeclarationActual]
            ) [ConstDecl]
          ) [ConstDeclarationTypeListSemi]
        ) [DeclarationListConst]
      ) [DeclarationListDeclaration]
      DeclarationListVar(
        VarDeclarationV(
          VarType(
            Type(
             int
            ) [Type]
          ) [VarType]
          VarDecl(
            IdentArray(
             niz1
            ) [IdentArray]
          ) [VarDecl]
        ) [VarDeclarationV]
      ) [DeclarationListVar]
    ) [DeclarationListDeclaration]
    DeclarationListVar(
      VarDeclarationV(
        VarType(
          Type(
           int
          ) [Type]
        ) [VarType]
        VarDecl(
          IdentArray(
           niz2
          ) [IdentArray]
        ) [VarDecl]
      ) [VarDeclarationV]
    ) [DeclarationListVar]
  ) [DeclarationListDeclaration]
  MethodDeclarationListMethod(
    MethodDeclarationListMethod(
      NoMethodDeclarationList(
      ) [NoMethodDeclarationList]
      VoidMethodDeclaration(
        MethodVoidName(
         func
        ) [MethodVoidName]
        MethodDeclarationActual(
          FormalParamsList(
            FormalParamListActual(
              FormalParamVar(
                Type(
                 int
                ) [Type]
                Ident(
                 a
                ) [Ident]
              ) [FormalParamVar]
              Type(
               char
              ) [Type]
              Ident(
               c
              ) [Ident]
            ) [FormalParamListActual]
          ) [FormalParamsList]
          NoDeclarationList(
          ) [NoDeclarationList]
          NoStatementList(
          ) [NoStatementList]
        ) [MethodDeclarationActual]
      ) [VoidMethodDeclaration]
    ) [MethodDeclarationListMethod]
    VoidMethodDeclaration(
      MethodVoidName(
       main
      ) [MethodVoidName]
      MethodDeclarationActual(
        NoFormalParamsList(
        ) [NoFormalParamsList]
        NoDeclarationList(
        ) [NoDeclarationList]
        StatementListStatement(
          NoStatementList(
          ) [NoStatementList]
          DesignatorStatementStatement(
            LeftDesignatorAssignExpression(
              LeftSideAssignVar(
                DesignatorOnly(
                 d
                ) [DesignatorOnly]
              ) [LeftSideAssignVar]
              AssignOpEqual(
                AssignOp(
                ) [AssignOp]
              ) [AssignOpEqual]
              ExpressionTermDef(
                MinusTerm(
                  TermFactor(
                    FactorDesignator(
                      DesignatorOnly(
                       c
                      ) [DesignatorOnly]
                    ) [FactorDesignator]
                  ) [TermFactor]
                ) [MinusTerm]
              ) [ExpressionTermDef]
            ) [LeftDesignatorAssignExpression]
          ) [DesignatorStatementStatement]
        ) [StatementListStatement]
      ) [MethodDeclarationActual]
    ) [VoidMethodDeclaration]
  ) [MethodDeclarationListMethod]
) [Program]
INFO  01:15:56,271 - ==================SEMANTICKA OBRADA==================
INFO  01:15:56,274 - Deklarisana promenljiva d(Var d: int, 0, 0 ) na liniji 5
INFO  01:15:56,274 - Deklarisana konstanta c(Con c: char, 97, 0 ) na liniji 6
ERROR 01:15:56,274 - Tip identifikatora konstante i tip izraza koji se dodeljuje se ne poklapaju
INFO  01:15:56,274 - Deklarisana konstanta cond(Con cond: bool , 1, 0 ) na liniji 7
INFO  01:15:56,274 - Deklarisan niz niz1(Var niz1: array, 1, 0 ) na liniji 8
INFO  01:15:56,274 - Deklarisan niz niz2(Var niz2: array, 2, 0 ) na liniji 9
INFO  01:15:56,274 - Obradjuje se funkcija func(Meth func: none, 0, 0 ) na liniji 11
INFO  01:15:56,274 - Obradjuje se funkcija main(Meth main: none, 0, 0 ) na liniji 15
INFO  01:15:56,274 - Koriscenje  d(Var d: int, 0, 0 ) na liniji 16
INFO  01:15:56,274 - Koriscenje  c(Con c: char, 97, 0 ) na liniji 16
ERROR 01:15:56,275 - Greska na liniji 16 : tip mora biti int
ERROR 01:15:56,275 - Greska na liniji 16 : nije moguca dodela2!
INFO  01:15:56,275 - ==================SEMANTICKA ANALIZA==============
INFO  01:15:56,275 - 2  methods in the program
INFO  01:15:56,275 - 1  global variables
INFO  01:15:56,283 - 2  global constants
INFO  01:15:56,284 - 2  global arrays
INFO  01:15:56,284 - 0  local variables in main
INFO  01:15:56,284 - 1  statements in main
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: bool, 0, 1 
Prog test302: notype, 0, 1 
   Var d: int, 0, 0 
   Con c: char, 97, 0 
   Con cond: bool, 1, 0 
   Var niz1: Arr of int, 1, 0 
   Var niz2: Arr of int, 2, 0 
   Meth func: notype, 0, 1 

   Meth main: notype, 0, 0 



ERROR 01:15:56,285 - Program nije semanticki ispravan!
